File: iteration1/TestGrade.java
public class TestGrade {
    public static void main(String[] args) {
        System.out.printf("Testing Grade.java\n");

        Grade grade1 = new Grade(95);

        System.out.println("\nTesting getGradeLetter() function...");
        if(grade1.getGradeLetter().equals("AA")){
            System.out.println("Test passed for 1.grade");
        }
        else{
            System.out.println("Test failed for 1.grade");
        }

        Grade grade2 = new Grade(55);
        if(grade2.getGradeLetter().equals("CC")){
            System.out.println("Test passed for 2.grade");
        }
        else{
            System.out.println("Test failed for 2.grade");
        }

        Grade grade3 = new Grade(0);
        if(grade3.getGradeLetter().equals("FF")){
            System.out.println("Test passed for 3.grade");
        }
        else{
            System.out.println("Test failed for 3.grade");
        }

        Grade grade4 = new Grade(110);
        if(grade4.getGradeLetter().equals("Invalid Grade")){
            System.out.println("Test passed for 4.grade");
        }
        else{
            System.out.println("Test failed for 4.grade");
        }

        Grade grade5 = new Grade(-50);
        if(grade5.getGradeLetter().equals("Invalid Grade")){
            System.out.println("Test passed for 5.grade");
        }
        else{
            System.out.println("Test failed for 5.grade");
        }
    }
}

File: iteration3/testClasses/test_grade.py
import unittest
from Grade import Grade


class TestGrade(unittest.TestCase):
    def test_is_retakable_grade(self):
        grade1 = Grade(45)
        self.assertTrue(grade1.is_retakable_grade())

    def test_is_not_retakable_grade(self):
        grade2 = Grade(60)
        self.assertFalse(grade2.is_retakable_grade())

    def test_is_not_passed_grade(self):
        grade1 = Grade(30)
        self.assertFalse(grade1.is_passed_grade())

    def test_is_passed_grade(self):
        grade2 = Grade(40)
        self.assertTrue(grade2.is_passed_grade())

    def test_get_letter_grade_AA(self):
        grade1 = Grade(95)
        self.assertEqual(grade1.get_letter_grade(), "AA")

    def test_get_letter_grade_FF(self):
        grade4 = Grade(20)
        self.assertEqual(grade4.get_letter_grade(), "FF")

    def test_get_invalid_grade(self):
        grade5 = Grade(105)
        self.assertEqual(grade5.get_letter_grade(), "Invalid Grade")


if __name__ == '__main__':
    unittest.main()